sansa-ml/sansa-ml-spark/src/main/scala/net/sansa_stack/ml/spark/utils/SPARQLQuery.scala

```diff
@@ -0,0 +1,189 @@
+package net.sansa_stack.ml.spark.utils
+
+import scala.collection.JavaConverters._
+
+import net.sansa_stack.query.spark.ontop.OntopSPARQLEngine
+import net.sansa_stack.rdf.common.partition.core.{RdfPartitionComplex, RdfPartitionerComplex}
+import net.sansa_stack.rdf.spark.partition.core.RdfPartitionUtilsSpark
+import org.apache.jena.graph.{Node, Triple}
+import org.apache.jena.query.Query
+import org.apache.jena.riot.Lang
+import org.apache.jena.sparql.core.Var
+import org.apache.jena.sparql.engine.binding.Binding
+import org.apache.jena.sparql.lang.ParserSPARQL11
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.param.ParamMap
+import org.apache.spark.ml.util.Identifiable
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql._
+import org.apache.spark.sql.types.{DataTypes, StructField, StructType}
+
+/**
+  * A SPARQL query transformer instance takes a SPARQL query string as
+  * constructor argument and can than be invoked (via the transform method) on
+  * a Dataset[Triple] object returning a DataFrame which contains one column
+  * for each SPARQL projection variable.
+  *
+  * The SPARQL query transformer only works with SELECT queries.
+  *
+  * @param queryString String containing the SPARQL select query to run on a
+  *                    Dataset[Triple] object
+  */
+abstract class SPARQLQuery(queryString: String) extends Transformer {
+  override val uid: String = Identifiable.randomUID("sparqlQuery")
+
+  protected val spark = SparkSession.builder().getOrCreate()
```

Comment:
do we need to setup here this crossjoin config to True? maybe session is set this way but what if it is not